### DISCUSSION TITLE
Define workflow programatically

### DISCUSSION THREAD

**Discussion by satishviswanathan** [elsa-workflows/elsa-core#4026] [Q&A]
## Raise signal ##

await _signaler.TriggerSignalAsync("userCreatedSignalReceived", user,null, user.Id, cancellationToken).ConfigureAwait(false);

## How to receive signal ##

how do I receive the signal. Is it my creating an activity with base class as SignalReceived. Is there a sample code for this ?

public class ItemCreateSignalReceivedEvent : SignalReceived
{
}

**Comment by sfmskywalker** [comment]
The SignalReceived activity wasn't designed for inheritance, so that won't work. Is there are reason you want to create a derived class instead of using SignalReceived directly?
